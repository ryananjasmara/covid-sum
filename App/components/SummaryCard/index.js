import React from 'react';
import {StyleSheet, View, Text} from 'react-native';
import WORDS from '../../configs/words';
import {thousandSeparator} from '../../helpers';
import {GRAY, ORANGE, RED, GREEN} from '../../configs/colors';
import {MoodBad, MoodGood, SentimentDissatisfied} from '../../assets/icons';

const SummaryCard = ({data}) => {
  const {confirmed, deaths, recovered} = data;

  return (
    <View style={styles.container}>
      <View style={styles.card}>
        <Text style={styles.categoriesTitle}>{WORDS.CONFIRMED}</Text>
        <SentimentDissatisfied />
        <Text style={styles.categoriesValues(ORANGE)}>
          {thousandSeparator(confirmed)}
        </Text>
      </View>
      <View style={styles.card}>
        <Text style={styles.categoriesTitle}>{WORDS.DEATHS}</Text>
        <MoodBad />
        <Text style={styles.categoriesValues(RED)}>
          {thousandSeparator(deaths)}
        </Text>
      </View>
      <View style={styles.card}>
        <Text style={styles.categoriesTitle}>{WORDS.RECOVERED}</Text>
        <MoodGood />
        <Text style={styles.categoriesValues(GREEN)}>
          {thousandSeparator(recovered)}
        </Text>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    justifyContent: 'space-evenly',
  },
  card: {
    padding: 16,
    backgroundColor: '#d9d9d9',
    justifyContent: 'center',
    alignItems: 'center',
    borderRadius: 10,
  },
  categoriesTitle: {
    color: GRAY,
  },
  categoriesValues: (color) => {
    return {
      color,
    };
  },
});

export default SummaryCard;
