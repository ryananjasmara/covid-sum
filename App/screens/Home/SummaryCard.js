import React from 'react';
import {StyleSheet, View, Text} from 'react-native';
import {CONFIRMED, DEATHS, RECOVERED} from '../../configs/words';

const SummaryCard = () => {
  return (
    <View style={styles.container}>
      <View style={styles.card}>
        <Text style={styles.categoriesTitle}>{CONFIRMED}</Text>
        <Text style={styles.categoriesValues('orange')}>10000</Text>
      </View>
      <View style={styles.card}>
        <Text style={styles.categoriesTitle}>{CONFIRMED}</Text>
        <Text style={styles.categoriesValues('orange')}>10000</Text>
      </View>
      <View style={styles.card}>
        <Text style={styles.categoriesTitle}>{CONFIRMED}</Text>
        <Text style={styles.categoriesValues('orange')}>10000</Text>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    justifyContent: 'space-between',
  },
  card: {
    padding: 16,
    backgroundColor: '#d9d9d9',
    justifyContent: 'center',
    alignItems: 'center',
  },
  categoriesTitle: {
    color: '#616161',
  },
  categoriesValues: (color) => {
    return {
      color,
    };
  },
});

export default SummaryCard;
