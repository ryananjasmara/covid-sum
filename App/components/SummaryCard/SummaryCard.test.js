import React from 'react';
import renderer from 'react-test-renderer';
import SummaryCard from './index';

describe('SummaryCard Component', () => {
  const initialProps = {
    data: {
      confirmed: 3116398,
      deaths: 217153,
      recovered: 928658,
    },
  };

  test('renders with data', () => {
    const tree = renderer.create(<SummaryCard {...initialProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders without data', () => {
    const tree = renderer.create(<SummaryCard />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
