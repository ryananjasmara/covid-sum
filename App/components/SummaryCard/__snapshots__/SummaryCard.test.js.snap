// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SummaryCard Component renders with data 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "justifyContent": "space-evenly",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#d9d9d9",
        "borderRadius": 10,
        "justifyContent": "center",
        "padding": 16,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#616161",
        }
      }
    >
      Confirmed
    </Text>
    <View>
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight="50"
        bbWidth="50"
        focusable={false}
        height="50"
        meetOrSlice={0}
        minX={0}
        minY={0}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            Object {
              "flex": 0,
              "height": 50,
              "width": 50,
            },
          ]
        }
        vbHeight={24}
        vbWidth={24}
        width="50"
      >
        <RNSVGGroup>
          <RNSVGPath
            d="M0 0h24v24H0V0z"
            fill={null}
            propList={
              Array [
                "fill",
              ]
            }
          />
          <RNSVGCircle
            cx="15.5"
            cy="9.5"
            fill={4294944000}
            propList={
              Array [
                "fill",
              ]
            }
            r="1.5"
          />
          <RNSVGCircle
            cx="8.5"
            cy="9.5"
            fill={4294944000}
            propList={
              Array [
                "fill",
              ]
            }
            r="1.5"
          />
          <RNSVGPath
            d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8zm0-3.5c.73 0 1.39.19 1.97.53.12-.14.86-.98 1.01-1.14-.85-.56-1.87-.89-2.98-.89-1.11 0-2.13.33-2.99.88.97 1.09.01.02 1.01 1.14.59-.33 1.25-.52 1.98-.52z"
            fill={4294944000}
            propList={
              Array [
                "fill",
              ]
            }
          />
        </RNSVGGroup>
      </RNSVGSvgView>
    </View>
    <Text
      style={
        Object {
          "color": "orange",
        }
      }
    >
      3.116.398
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#d9d9d9",
        "borderRadius": 10,
        "justifyContent": "center",
        "padding": 16,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#616161",
        }
      }
    >
      Deaths
    </Text>
    <View>
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight="50"
        bbWidth="50"
        focusable={false}
        height="50"
        meetOrSlice={0}
        minX={0}
        minY={0}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            Object {
              "flex": 0,
              "height": 50,
              "width": 50,
            },
          ]
        }
        vbHeight={24}
        vbWidth={24}
        width="50"
      >
        <RNSVGGroup>
          <RNSVGPath
            d="M0 0h24v24H0V0z"
            fill={null}
            propList={
              Array [
                "fill",
              ]
            }
          />
          <RNSVGPath
            d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8zm3.5-9c.83 0 1.5-.67 1.5-1.5S16.33 8 15.5 8 14 8.67 14 9.5s.67 1.5 1.5 1.5zm-7 0c.83 0 1.5-.67 1.5-1.5S9.33 8 8.5 8 7 8.67 7 9.5 7.67 11 8.5 11zm3.5 3c-2.33 0-4.31 1.46-5.11 3.5h10.22c-.8-2.04-2.78-3.5-5.11-3.5z"
            fill={4294901760}
            propList={
              Array [
                "fill",
              ]
            }
          />
        </RNSVGGroup>
      </RNSVGSvgView>
    </View>
    <Text
      style={
        Object {
          "color": "red",
        }
      }
    >
      217.153
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#d9d9d9",
        "borderRadius": 10,
        "justifyContent": "center",
        "padding": 16,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#616161",
        }
      }
    >
      Recovered
    </Text>
    <View>
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight="50"
        bbWidth="50"
        focusable={false}
        height="50"
        meetOrSlice={0}
        minX={0}
        minY={0}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            Object {
              "flex": 0,
              "height": 50,
              "width": 50,
            },
          ]
        }
        vbHeight={24}
        vbWidth={24}
        width="50"
      >
        <RNSVGGroup>
          <RNSVGPath
            d="M0 0h24v24H0V0z"
            fill={null}
            propList={
              Array [
                "fill",
              ]
            }
          />
          <RNSVGPath
            d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8zm3.5-9c.83 0 1.5-.67 1.5-1.5S16.33 8 15.5 8 14 8.67 14 9.5s.67 1.5 1.5 1.5zm-7 0c.83 0 1.5-.67 1.5-1.5S9.33 8 8.5 8 7 8.67 7 9.5 7.67 11 8.5 11zm3.5 6.5c2.33 0 4.31-1.46 5.11-3.5H6.89c.8 2.04 2.78 3.5 5.11 3.5z"
            fill={4278222848}
            propList={
              Array [
                "fill",
              ]
            }
          />
        </RNSVGGroup>
      </RNSVGSvgView>
    </View>
    <Text
      style={
        Object {
          "color": "green",
        }
      }
    >
      928.658
    </Text>
  </View>
</View>
`;

exports[`SummaryCard Component renders without data 1`] = `null`;

exports[`renders correctly 1`] = `null`;
